backend/modules/assistant/ito/difference_assistant.py

```diff
@@ -0,0 +1,287 @@
+
+        content = generate_authorisation_report(diff_df)
+
+        original_filename = "Classic Ingredient Comparison"
+        file_description = "Difference Report"
+        processed_file = self.create_and_upload_processed_file(
+            content,
+            original_filename=original_filename,
+            file_description=file_description,
+        )  # FIXME change name
+        file_to_upload = processed_file["file_to_upload"]
+        new_filename = processed_file["new_filename"]
+
+        # Email the file if required
+        if self.input.outputs.email.activated:
+            await self.send_output_by_email(
+                file_to_upload,
+                new_filename,
+                "Summary",
+                f"{file_description} of {original_filename}",
+                brain_id=(
+                    self.input.outputs.brain.value
+                    if self.input.outputs.brain.activated
+                    and self.input.outputs.brain.value
+                    else None
+                ),
+            )
+
+        # Reset to start of file before upload
+        file_to_upload.file.seek(0)
+
+        # Upload the file if required
+        if self.input.outputs.brain.activated:
+            await upload_file(
+                uploadFile=file_to_upload,
+                brain_id=self.input.outputs.brain.value,
+                current_user=self.current_user,
+                chat_id=None,
+            )
+
+        return {"message": f"{file_description} generated successfully"}
+
+
+def generate_authorisation_report(df: pd.DataFrame):
+    # Ensure the dataframe has the required columns
+    if "name" not in df.columns or "decision" not in df.columns:
+        raise ValueError("DataFrame must contain 'name' and 'decision' columns")
+
+    # Extract forbidden and to avoid ingredients
+    forbidden_ingredients = df[df["decision"] == "Forbidden"][
+        ["name", "detailed_answer"]
+    ].values.tolist()
+    to_avoid_ingredients = df[df["decision"] == "To Avoid"][
+        ["name", "detailed_answer"]
+    ].values.tolist()
+
+    # Create the markdown content
+    markdown_content = "# Coup de Pates\n\n"
+    markdown_content += "## Rapport d'analyse\n\n"
+
+    if forbidden_ingredients:
+        markdown_content += "### Ingrédients utilisés **Interdits**\n\n"
+        for ingredient, detail in forbidden_ingredients:
+            markdown_content += f"**{ingredient}**: *{detail}*\n\n"
+    else:
+        markdown_content += "### Ingrédients utilisés **Interdits**\n- Aucun\n\n"
+
+    markdown_content += "\n"
+
+    if to_avoid_ingredients:
+        markdown_content += "### Ingrédients utilisés **à éviter**\n"
+        for ingredient, detail in to_avoid_ingredients:
+            markdown_content += f"**{ingredient}** : *{detail}*\n\n"
+    else:
+        markdown_content += "### To Avoid Ingredients\n- None\n\n"
+
+    return markdown_content
+
+
+def difference_inputs():
```

Comment:
small naming change `difference_outputs`

Reply:
Called like that in list assistant / assistant_routes